fs/tpe.c

```diff
@@ -0,0 +1,112 @@
+/*
+ *  linux/fs/tpe.c
+ *
+ */
+#include <linux/kernel.h>
+#include <linux/sched.h>
+#include <linux/file.h>
+#include <linux/fs.h>
+#include <linux/uidgid.h>
+#include <linux/tpe.h>
+#include <linux/ratelimit.h>
+#include <linux/limits.h>
+#include <linux/dcache.h>
+#include <linux/cred.h>
+#include <linux/slab.h>
+
+#define TPE_GLOBAL_UID(x) from_kuid_munged(&init_user_ns, (x))
+#define TPE_GLOBAL_GID(x) from_kgid_munged(&init_user_ns, (x))
+#define tpe_is_global_root(x) uid_eq((x), GLOBAL_ROOT_UID)
+#define tpe_is_global_nonroot(x) (!uid_eq((x), GLOBAL_ROOT_UID))
+#define tpe_is_global_nonroot_gid(x) (!gid_eq((x), GLOBAL_ROOT_GID))
+
+int security_tpe = IS_ENABLED(CONFIG_SECURITY_TPE);
```

Comment:
These can't be read-only yet but they might as well be defined as int security_tpe __read_mostly to at least serve as a reminder to switch them.